### PR TITLE
Added tests for RealtimeSegmentValidationManager

### PR DESCRIPTION
**Pull request by mcvsubbu**
Removed the dropping of realtime table and setting it up again. Relying on
the realtime table being created ahead of time

### CODE REVIEW

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 1]
```diff
@@ -73,6 +73,8 @@
         "controller.validation.frequencyInSeconds";
     private static final String OFFLINE_SEGMENT_INTERVAL_CHECKER_FREQUENCY_IN_SECONDS =
         "controller.offline.segment.interval.checker.frequencyInSeconds";
+    private static final String SEGMENT_VALIDATION_INITIAL_DELAY_IN_SECONDS =
```
Comment (npawar) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: should this be called REALTIME_SEGMENT_VALIDATION_INITIAL_DELAY_IN_SECONDS to match other variable conventions(like REALTIME_SEGMENT_VALIDATION_FREQUENCY_IN_SECONDS)

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 2]
```diff
@@ -463,6 +465,10 @@ public long getBrokerResourceValidationInitialDelayInSeconds() {
         getPeriodicTaskInitialDelayInSeconds());
   }
 
+  public void setSegmentLevelValidationInitialDelayInSeconds(long validationInitialDelayInSeconds) {
```
Comment (npawar) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: setRealtimeSegmentValidationInitialDelayInSeconds ?

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 3]
```diff
@@ -635,6 +641,9 @@ public void setOfflineSegmentIntervalCheckerInitialDelayInSeconds(long initialDe
   }
 
   public long getPeriodicTaskInitialDelayInSeconds() {
+    if (containsKey(ControllerPeriodicTasksConf.SEGMENT_LEVEL_VALIDATION_INTERVAL_IN_SECONDS)) {
```
Comment (npawar) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: This one is the generic PeriodicTask initial delay. The segment level validation seconds shouldn't be applied here. 
Maybe we need another on getRealtimeSegmentValidationInitialDelay()

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java [hunk 4]
```diff
@@ -635,6 +641,9 @@ public void setOfflineSegmentIntervalCheckerInitialDelayInSeconds(long initialDe
...
+      return getLong(ControllerPeriodicTasksConf.SEGMENT_LEVEL_VALIDATION_INTERVAL_IN_SECONDS);
+    }
```
Comment (npawar) [inline pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: getLong(key, defaultValue) should reduce this to 1 line

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ControllerPeriodicTasksIntegrationTests.java [hunk 5]
```diff
@@ -499,6 +492,35 @@ public void testOfflineSegmentIntervalChecker()
...
+    Assert.assertTrue(controllerMetrics.getValueOfGlobalGauge(ControllerGauge.PERIODIC_TASK_NUM_TABLES_PROCESSED,
+        "RealtimeSegmentValidationManager") > 0);
+    RealtimeSegmentValidationManager validationManager = _controllerStarter.getRealtimeSegmentValidationManager();
+    ValidationMetrics validationMetrics = validationManager.getValidationMetrics();
+    // Make sure we processed the realtime table to get the total document count. Should have been done the first
+    // time RealtimeSegmentValidationManager ran on the default realtime table.
+    Assert.assertTrue(validationMetrics.getValueOfGauge(ValidationMetrics.makeGaugeName(getDefaultRealtimeTableName(),
+        "TotalDocumentCount")) > 0);
+  }
+
```
Comment (npawar) [inline pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ControllerPeriodicTasksIntegrationTests.java]: should we drop the realtime table in the after() for this test, or in the teardown() ?

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java (hunk 1 above)
Comment (mcvsubbu) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: All segment-level validation are done with a single call to get a random back-off. Therefore it makes sense to have a single variable. RealtimeSegemntValidationManager calls getPeriodicTasksInitialDelay(). Do we really want to introduce(even) more configs just for test sake?

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java (hunk 2 above)
Comment (mcvsubbu) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: See comment above regarding common configuration.

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java (hunk 3 above)
Comment (mcvsubbu) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: No. The code calls this method  to get the initial delay

File: pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java (hunk 4 above)
Comment (mcvsubbu) [reply pinot-controller/src/main/java/org/apache/pinot/controller/ControllerConf.java]: Disagree. We don't want users configuring this value, or provide a default if they don't. We want to choose random delay all the time except for tests.

File: pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ControllerPeriodicTasksIntegrationTests.java (hunk 5 above)
Comment (mcvsubbu) [reply pinot-integration-tests/src/test/java/org/apache/pinot/integration/tests/ControllerPeriodicTasksIntegrationTests.java]: Not necessary. We are tearing down the cluster at the end anyway.

**Pull request merged**